app/scripts/controllers/transactions/index.js

```diff
@@ -542,6 +542,15 @@ class TransactionController extends EventEmitter {
     })
     this.pendingTxTracker.on('tx:failed', this.txStateManager.setTxStatusFailed.bind(this.txStateManager))
     this.pendingTxTracker.on('tx:confirmed', (txId) => this.confirmTransaction(txId))
+    this.pendingTxTracker.on('tx:dropped', (txId) => {
+      const txMeta = this.txStateManager.getTx(txId)
+      if (!txMeta.dropBlockBuffer) {
```

Comment:
where does this get set?

Reply:
the line bellow

Reply:
i meant what is the case in which this code path is not always true?
i'm confused on the flow, it seems:
on tx:dropped, update the txMeta with a tx:drop-update
then on another dropped event for the same tx, we setTxStatusDropped?

are we expecting multiple dropped events for the same tx?